### PR TITLE
Feature/fix broken images

### PR DESCRIPTION
**Pull request by MSECode**
This PR add some fixes to the basic images. 
Specifically we introduced a fix to make working the image `superbuild-icubhead-withuser` and the images that uses `robometry`.
Moreover we improved the workflows and clean the `Dockerfile`  of the images removing building of `appsAway` repository that is not needed anymore and fixing some of the instructions to make the `DockerFile` cleaner.

Major changes made:
- remove `https://github.com/icub-tech-iit/appsAway.git` from image build since not needed anymore
- cleanup of arg variable
- cleanup of superbuild tags keeping just the needed ones
- check of multistage build
- move superbuild-gazebo to stale images since it needs revision
- removed `YCM_EP_INSTALL_DIR`
- change superbuild installation folder to default folder `robotology-superbuild/build/install`

### CODE REVIEW

**Comment by MSECode** [comment]
Since I'm encountering other errors during the build process of the images that need to re-build the `robotology-superbuild` starting from a not empty cache of it, such as the image `superbuild-icubhead-robometry` that uses `superbuild-icubhead-withuser` as starting image, I'll keep this PR in draft until those errors gets cleaned. 
Anyways, this problem needs to be investigated more considering that it is not just an issue in the building of the `children` images but you can have the same issue if you desire to run a container based on that image and you desire to re-build the `superbuild`, as an example when you need to test your code and then you need to build in devel mode. 
In particular, the error I'm now talking about is shown by the screen below. I've encountered it while re-build the `robotology-superbuild` starting from the image `superbuild-icubhead-withuser` and enabling the option `ROBOTOLOGY_ENABLE_DYNAMICS`.
I've got the same exact issue either if setting `CMAKE_INSTALL_PREFIX` to the standard path: `robotology-superbuild/build/install` or to another custom directory. Thinking that it could have been a cmake version problem I've also tried to generate the base image not using cmake 3.22.1, which is the one given by `ubuntu:jammy` by default, but removing it and re-installing cmake and cmake-guid version 3.29.0 during the generation of the image. In both cases the outcome was the same.

image

**Comment by valegagge** [comment]
Hi @MSECode , 
thanks for your contribution!
Can you summarize the major changes in bullet points hereafter?

In addition please take into account this issue https://github.com/icub-tech-iit/docker-deployment-images/issues/37 during your tests.

**Comment by MSECode** [comment]
> Hi @MSECode , thanks for your contribution! Can you summarize the major changes in bullet points hereafter?
> 
> In addition please take into account this issue #37 during your tests.

Update the PR body

**Review by valegagge — APPROVED** [review]
@MSECode already tested the image. All changes are fine for me. Let's merge!
